scrapy/core/scheduler.py

```diff
@@ -296,6 +296,15 @@ def __init__(
         self._smqclass: type[BaseQueue] | None = self._get_start_queue_cls(
             crawler, "MEMORY"
         )
+        if (
+            self.crawler
+            and self.crawler.settings.getint("CONCURRENT_REQUESTS_PER_IP") != 0
```

Comment:
I think it may make sense to use `get` here with a sentinel as default, to trigger the warning even if the setting is set to `0` or even `None`:

```python
_NOT_SET = object()
...
self.crawler.settings.get("CONCURRENT_REQUESTS_PER_IP", _NOT_SET) is not _NOT_SET
```

Reply:
Refactored, lmk what you think.